modules/sdk-coin-sui/src/lib/iface.ts

```diff
@@ -57,11 +137,67 @@ export interface PayTx {
   amounts: number[];
 }
 
-export interface SuiTransaction {
+export interface MoveCallTx {
+  package: SuiObjectRef;
+  module: string;
+  function: string;
+  typeArguments: TypeTag[];
+  arguments: SuiJsonValue[];
+}
+
+export interface RequestAddDelegationTx {
+  coins: SuiObjectRef[];
+  amount: number;
+  validatorAddress: SuiAddress;
+}
+
+export interface SuiTransaction<T = PayTx | MoveCallTx> {
```

Comment:
Just to understand this: why you here are not adding the RequestAddDelegationTx to the supported SUITransaction types here?

Reply:
I should rename `RequestAddDelegationTx` to just `RequestAddDelegation`  or `RequestAddDelegationDTO` it's an interface used to construct a transaction as parameter to `requestAddDelegation(addDelegationTx: RequestAddDelegationTx)` just to avoid passing arguments separately in method 

alternative will be `requestAddDelegation(coins, amout, validatorAddress)

as the result a MoveCallTx will be constructed 
```
    this._moveCallTx = {
      package: SUI_PACKAGE,
      module: ModulesNames.SuiSystem,
      function: MethodNames.RequestAddDelegationMulCoin,
      typeArguments: [],
      arguments: [
        SUI_SYSTEM_STATE_OBJECT,
        addDelegationTx.coins,
        addDelegationTx.amount,
        addDelegationTx.validatorAddress,
      ],
    }
    ```

Reply:
`MoveCallTx` it's the which will be eventually broadcasted to chain

Reply:
Thanks for clarifty